CMakeLists.txt

```diff
@@ -20,12 +24,43 @@ endif ()
 
 add_compile_options(-Wno-deprecated)  # This is just to avoid some spammy warnings
```

Comment:
[fix]/[plan] Can remove this now that it is set per-target.

Reply:
https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/16/commits/135c8b8aa252f19af13afced0e83fc1088db37fc

(part of another fixup)